sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -126,7 +126,10 @@ private[sql] class DataFrameImpl protected[sql](
     logicalPlan.isInstanceOf[LocalRelation]
   }
 
-  override def show(): Unit = {
+  /**
+   * Internal API for Python
+   */
+  private[sql] def samples(): String = {
```

Comment:
samples is a bad name for this. since this is internal, we can create a longer name. maybe showString?